### PR TITLE
Wizard spell: Bound Object

### PR DESCRIPTION
**Pull request by Sprok0**
wizbox
This is a wizard spell that lets a wizard bind any non-anchored object, then teleport it to their location. Items will be put into the wizard's hands if possible, otherwise will be dropped onto the wizard's tile. The spell can be upgraded to be able to grab most anchored objects. The bound object can be deselected and a new one bound at any time by middle clicking the spell icon. There's also an option to get a specific "unbind" spell when you buy this if you don't have a middle mouse button for whatever reason. I don't know if this is worth a full 5 spell points, but there's probably some dumb things that can be done with it. Added a one-use spellbook for the bundle.
wizshard
Thanks to unid15 for the middle-click code, I'd never understand it otherwise

### CODE REVIEW

**Comment by Aurx** [comment]
There are people without mice that can middle click.  
Edit: Because there's apparent confusion regarding my comment: Current description indicates remarking would be a feature only available to people able to middle click.  Right now, all player features accessed via middle click can also be accessed in other ways.  If you do not provide another way to remark, you are denying functionality to people who cannot middleclick.

**Comment by icantthinkofanameritenow** [comment]
What's the first gif showing?

**Comment by Sprok0** [comment]
Throwing a toolbox at a catbeast, warping it back into your hand, and repeating

**Comment by icantthinkofanameritenow** [comment]
Ah, wasn't sure. 

Well, I guess it could be used like the subspace tunneler; have you considered making it work on machines too?

**Comment by Sprok0** [comment]
It works on any /obj that isn't anchored, does the tunneler work on anchored things?

**Comment by icantthinkofanameritenow** [comment]
It does when it's given a natural blueapace crystal and a super matter bin.

**Comment by coldcola** [comment]
maybe a fully upgraded Recall spell can move anchored objects? 
could be cool.

**Comment by SonixApache** [comment]
Wouldn't that let you throw the AI around?

**Comment by Exxion** [comment]
A wizard can already command the AI to unbolt itself then use this on it

**Comment by Sprok0** [comment]
The spell only works on objects, the AI is a mob

**Comment by Arthurdentist** [comment]
first use I can think of would be marking one of the staffs I.E. staff of change.

**Comment by clusterfack** [comment]
This seems like telekinetic grab, when I hear mark and recall I think of the teleportation spell from morrowind.

**Comment by Durgen** [comment]
can you recall a singulo?

**Comment by Shadowmech88** [comment]
Even if you could it'd kill you when it landed on you.

**Comment by ScrambledSteak** [comment]
> about to die
> recall the singulo to your location
> blaze of glory

**Comment by Sprok0** [comment]
Added a middle-click alternative, a one-use spellbook for the bundle, and upgrading power now allows most anchored objects to be marked.

I'm not really attached to the name, and mark/recall should probably be saved for the morrowind spell, anyone have a better name?

**Comment by AI-Is-Rogue** [comment]
How does "Summon Object" sound?

**Comment by PJB3005** [comment]
That sounds like you'd spawn an object from thin air, not that you'd teleport one.

**Comment by PJB3005** [comment]
How does "bound item" sound?

**Comment by AI-Is-Rogue** [comment]
Or "Summon bound object".

**Comment by Sprok0** [comment]
I went with "bound object"

**Comment by clusterfack** [comment]
Neat 👍

**Comment by Intigracy** [comment]
Slap a changelog on it.

File: code/modules/spells/targeted/recall.dm [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		/obj/machinery/door_timer,								//brig cell timers
+		/obj/structure/closet/secure_closet/brig,				//brig cell closets
+		/obj/machinery/disposal,								//disposal bins
+		/obj/machinery/light,									//light bulbs and tubes
+		/obj/machinery/sleep_console,							//sleeper consoles
+		/obj/machinery/sleeper,									//sleepers
+		/obj/machinery/body_scanconsole,						//body scanner consoles
+		/obj/machinery/bodyscanner,								//body scanners
+		/obj/machinery/media/receiver/boombox/wallmount,		//sound systems
+		/obj/machinery/keycard_auth,							//keycard authentication devices
```
Comment (Intigracy) [inline code/modules/spells/targeted/recall.dm:49]: I'd say singularities should be on here, but then again it's pretty much an instant kill if the wizard pops it in on themselves.

Shards though?

Comment (Sprok0) [reply code/modules/spells/targeted/recall.dm:49]: Shards will irradiate the shit out of the wizard if they grab it, and singularities are anchored so they'll need to spend two spell points on what is basically a suicide spell

**Pull request merged**